python/lsst/meas/astrom/astrometry.py

```diff
@@ -57,6 +59,18 @@ class AstrometryConfig(RefMatchConfig):
         default=0.001,
         min=0,
     )
+    doMagnitudeOutlierRejection = pexConfig.Field(
+        dtype=bool,
+        doc=("If True then a rough zeropoint will be computed from matched sources "
+             "and outliers will be rejected in the iterations."),
+        default=False,
+    )
+    magnitudeOutlierRejectionNSigma = pexConfig.Field(
+        dtype=float,
+        doc=("Number of sigma (measured from the distribution) in magnitude "
+             "for a potential match to be rejected during iteration."),
+        default=3.0,
```

Comment:
Instead of two config fields, could you just have the NSigma one and make it `optional` with `default=None`? That way, all the user has to do is set the value they want, and your `if` below just checks for whether that is `None`.

Reply:
I'm going to push back against this one.  Of all of the idiosyncrasies of the stack, I _love_ the `do*` pattern which makes it very clear which options are turned on and which are not.

Reply:
Based on a brief survey on #dm-science-pipelines, it looks like several others prefer your approach, too.